### PR TITLE
Change syntax for unit tuples

### PR DESCRIPTION
**Pull request by ajreynol**
Also fixes previous documentation for tuples.

This is in preparation for the alf proof backend.  The current use of `Tuple` to denote both unit and non-unit tuples is highly ambiguous. This makes `UnitTuple` the new smt2 syntax for unit tuples.  

The PR currently adds support but does not delete support for the old syntax (both can be used), although I am open to changing the syntax to the new version permanently.

This PR also adds several missing cases of smtKindString.

### CODE REVIEW

File: docs/theories/sets-and-relations.rst [hunk 1]
```diff
@@ -161,7 +161,11 @@ More details can be found in :cite:`MengRTB17`.
...
+| Tuple Constructor    | ``(tuple <Term_1>, ..., <Term_n>)``          | ``Term t = solver.mkTuple({Term_1>, ..., <Term_n>});``                             |
++----------------------+----------------------------------------------+------------------------------------------------------------------------------------+
+| Unit Tuple Sort      | ``UnitTuple``                                | ``Sort s = solver.mkTupleSort({});``                                               |
++----------------------+----------------------------------------------+------------------------------------------------------------------------------------+
+| Unit Tuple           | ``tuple.unit``                               | ``Term t = solver.mkTuple({});``                                                   |
 +----------------------+----------------------------------------------+------------------------------------------------------------------------------------+
 | Tuple Selector       | ``((_ tuple_select i) t)``                   | ``Sort s = solver.mkTupleSort(sorts);``                                            |
```
Comment (mudathirmahgoub) [inline docs/theories/sets-and-relations.rst:170]:
```suggestion
| Tuple Selector       | ``((_ tuple.select i) t)``                   | ``Sort s = solver.mkTupleSort(sorts);``                                            |
```

Comment (ajreynol) [reply docs/theories/sets-and-relations.rst:170]: Good catch

**Comment by mudathirmahgoub** [comment]
Removing the old syntax would break others code like [here](https://github.com/CVC4/org.alloytools.alloy/blob/merge/SMT-Extension/src/main/java/edu/uiowa/smt/smtAst/TupleSort.java) but keeping both is harder to maintain. May be it is better to remove old syntax in a future PR

**Comment by ajreynol** [comment]
Considering eliminating the old syntax already.

**Review by aniemetz — APPROVED** [review]
LGTM. I think we can go ahead and remove the old syntax, yes.

**Pull request merged**